Assets/Scripts/Game/StringTableCSVParser.cs

```diff
@@ -84,37 +84,36 @@ public class StringTableCSVParser
     /// <returns>KeyValuePair for each row.</returns>
     static KeyValuePair<string, string>[] ParseCSVRows(string csvText)
     {
-        // Regex pattern from https://gist.github.com/awwsmm/886ac0ce0cef517ad7092915f708175f
-        const string linePattern = "(?:,|\\n|^)(\"(?:(?:\"\")*[^\"]*)*\"|[^\",\\n]*|(?:\\n|$))";
+        // Regex pattern inspired by https://gist.github.com/awwsmm/886ac0ce0cef517ad7092915f708175f
+        // but without the exponential behavior
+        const string linePattern = "(?:\\n|^)([^\",\\n]*),((?:\"[^\"]*\")+|[^\",\\n]*)";
 
         // Split source CSV based on regex matches
-        char[] trimChars = { '\r', '\n', '\"', ',' };
-        List<KeyValuePair<string, string>> rows = new List<KeyValuePair<string, string>>();
-        string[] matches = (from Match m in Regex.Matches(csvText, linePattern, RegexOptions.ExplicitCapture) select m.Groups[0].Value).ToArray();
-        int pos = 0;
-        while (pos < matches.Length)
-        {
-            if (pos + 1 == matches.Length)
-            {
-                // Exit if no valid pair at end of csv (likely an empty line at end of source data)
-                break;
-            }
-            string key = matches[pos++].Trim(trimChars);
-            string value = matches[pos++].Trim(trimChars);
-            value = value.Replace("\"\"", "\""); // Replace escaped quotes in value with single quote marks
-            KeyValuePair<string, string> kvp = new KeyValuePair<string, string>(key, value);
-            rows.Add(kvp);
-        }
-
+        char[] trimChars = { '\r', '\n' };
+        List<KeyValuePair<string, string>> rows = (from Match m in
+            Regex.Matches(csvText, linePattern)
+            select new KeyValuePair<string, string>(
+                m.Groups[1].Value.Trim(trimChars),
+                UnescapeCSVvalue(m.Groups[2].Value).Trim(trimChars)
+                )
+            ).ToList();
         // Remove first row if it contains "Key" as key and "Value" as value
         // This is the expected header row but doesn't need to be present
         // First row will be accepted if any other key/value pair is present instead
         if (rows.Count > 0 && rows[0].Key == keyString && rows[0].Value == valueString)
             rows.RemoveAt(0);
-
         return rows.ToArray();
     }
 
+    static string UnescapeCSVvalue(string value) {
+        if (value.Length > 0 && value[0] == '"')
+        {
+            return value.Substring(1, value.Length - 2)
```

Comment:
Maybe we should check and report an error if the last character is not also a quote.

Reply:
No, because that's already guaranteed by the regex that if it starts with a quote, you're in the case it ends with a quote too

Reply:
Same goes for unescaping quotes only in this case by the way; If the string does not start with a quote, then it cannot contain any quote.